src/Compilers/CSharp/Portable/Symbols/Source/TypeParameterConstraintClause.cs

```diff
@@ -52,6 +54,19 @@ internal sealed class TypeParameterConstraintClause
             ImmutableArray<TypeConstraintSyntax> typeConstraintsSyntax,
             ImmutableArray<TypeParameterConstraintClause> otherPartialDeclarations)
         {
+#if DEBUG
+            switch (constraints & (TypeParameterConstraintKind.NullableReferenceType | TypeParameterConstraintKind.NotNullableReferenceType))
+            {
+                case TypeParameterConstraintKind.None:
+                case TypeParameterConstraintKind.ReferenceType:
+                case TypeParameterConstraintKind.NullableReferenceType:
+                case TypeParameterConstraintKind.NotNullableReferenceType:
+                    break;
+                default:
+                    ExceptionUtilities.UnexpectedValue(constraints); // This call asserts.
```

Comment:
>// This call asserts. [](start = 69, length = 21)

nit: seems simpler to do `throw ExceptionUtilities.UnexpectedValue(...);` as we do everywhere else

Reply:
> seems simpler to do throw ExceptionUtilities.UnexpectedValue(...); as we do everywhere else

I do not find this simpler and we do use this method for asserts in several places in the codebase, this is not a unique instance. 

---
In reply to: [239968757](https://github.com/dotnet/roslyn/pull/31607#discussion_r239968757) [](ancestors = 239968757)